### ISSUE TITLE
http proxy in proxy_config not works

### ISSUE THREAD

**Issue by tvytlx** [wkeeling/selenium-wire#27]
I have some questions

https://github.com/wkeeling/selenium-wire/blob/5ba4e30c67d3451da5f9e2d04b93db156c2047f3/seleniumwire/proxy/proxy2.py#L239-L247

`Proxy-Authorization` headers didn't pass to anything when .the `proxy_type` is not https?

then I try to add this line `conn.set_tunnel(netloc, headers=headers)` in the else condition, but it still not works, because

https://github.com/python/cpython/blob/8c349565e8a442e17f1a954d1a9996847749d778/Lib/http/client.py#L894-L897

it looks like the http tunnel method in the `http.client` lib only send `CONNECT` to negotiate with the proxy server, maybe we also need tunneling without using `CONNECT`, so that we can handle http request to the proxy server? (I'm not sure about that, need further discussion)

**Comment by wkeeling** [comment]
@tvytlx  thanks for raising this.

I assume that the `Proxy-Authorization` header is only supported in the https scenario due to the fact that `Basic` authentication is used which uses clear (base64 encoded) text. Although that assumption may need confirmation from the maintainer of the original `proxy2.py` library (from [here](https://github.com/inaz2/proxy2)). 

I'll do some of my own investigation into whether we can send the `Proxy-Authorization` header over http. In the meantime, you may be able to use Selenium Wire's `header_overrides` to set the header directly from your test. For example:
```

import base64

auth = '%s:%s' % (username, password)
driver.header_overrides = {
    'Proxy-Authorization': 'Basic ' + base64.b64encode(auth.encode('latin-1')).decode(
                        'latin-1')
}

```
That's admittedly not ideal, but may be a reasonable workaround for the time being.

**Comment by wkeeling** [comment]
It seems that apart from not sending the `Proxy-Authorization` header for http requests, Selenium Wire is also not sending requests correctly to the proxy when the protocol is http. It incorrectly assumes that a CONNECT tunnel is in place, and so sends a server-relative path without the protocol and host parts. This causes the proxy to respond with a 400 Bad Request.

For the http protocol it would be preferable to use the full path (protocol + host + path) when making a request rather than using a CONNECT tunnel. This is because [not all proxy servers support tunnelling or limit it to port 443](https://developer.mozilla.org/en-US/docs/Web/HTTP/Proxy_servers_and_tunneling#HTTP_tunneling). 

So we need to:

1) Ensure that Selenium Wire sends the full path when a http proxy is configured.
2) Ensure that Selenium Wire sends the `Proxy-Authorization` header when a http proxy is configured with authentication.

**Comment by wkeeling** [comment]
This has been fixed in master. Selenium Wire will now support authenticated proxies over HTTP without requiring a CONNECT tunnel. Available in the next release.

**Comment by wkeeling** [comment]
@tvytlx This fix is now available in version 1.0.2.
